### PR TITLE
Encoding guesser

### LINKED ISSUE #1237: [feature] Implement guesser for document encoding
**Is your feature request related to a problem? Please describe.**

#1178 allowed us to send compressed documents for ingestion. At the moment the document encoding must be explicitly declared. It would be really good to have a guessers for supported encoding formats, so we can deal with the cases when users don't provide this information.

**Describe the solution you'd like**

Find a good encoding guesser in golang that can detect most of the popular protocols like gzip, bzip2, zstd, ... Or implement one if it does not exists yet. From the brief research I did on the topic, it should be possible to guess the type for examining first 512 bytes of the document.

### PR DESCRIPTION
**Pull request by m-brophy**
# Description of the PR

This PR enables guac to detect the encoding of a compressed file from the internal bytes of the document, for use if this information isn't supplied separately

Fixes #1237 

`http.DetectContentType` is used but this does not handle `bz2` or `zst` encoding so this capability has been added

these docs were used as a reference for detection of these types: https://www.ietf.org/rfc/rfc8878.txt and https://www.ietf.org/rfc/rfc5655.txt

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [x] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: pkg/handler/processor/process/process.go [hunk 1]
```diff
@@ -251,3 +271,31 @@ func decompressDocument(i *processor.Document, reader io.Reader) error {
...
+		testBytes, err := reader.Peek(16)
+		if err != nil {
+			return "", err
+		}
+
+		//find the content mime-type from the first few bytes
+		contentType := http.DetectContentType(i.Blob)
+		//octet-stream is the default meaning that no encoding has been found
+		if contentType == "application/octet-stream" {
+			if testBytes[0] == 0x42 && testBytes[1] == 0x5A && testBytes[2] == 0x68 {
```
Comment (lumjjb) [inline pkg/handler/processor/process/process.go:291]: nit: can we create a function for checking the header and maybe add a link out/comment on what they check is doing (i.e. checking the file header, according to RFC xyz).

**Review by lumjjb — COMMENTED** [review]
minor nits, can we refactor the guesser to its own package? else LGTM!

**Comment by lumjjb** [comment]
Hey @m-brophy ! Just checking back on this! The PR looks almost ready to merge with some minor nits.

**Comment by m-brophy** [comment]
> Hey @m-brophy ! Just checking back on this! The PR looks almost ready to merge with some minor nits.

Thanks @lumjjb I'll look to get this done asap

**Comment by m-brophy** [comment]
Hi @lumjjb, changes made, branch rebased, CI tests all successful if you want to take another look now.

**Comment by m-brophy** [comment]
Hey @lumjjb I just rebased this ready to merge if you're happy to do that

**Review by lumjjb — APPROVED** [review]
LGTM! Thanks @m-brophy

**Review by jeffmendoza — APPROVED** [review]
Addressed some minor style nits.

**Pull request merged**